### PR TITLE
AD container size packages

### LINKED ISSUE #26300: Allow setting of AD container size for NCRC conda binaries
## Reason

NCRC conda binaries of MOOSE-based apps are currently configured with the default AD container size. Some inputs packaged with some of these applications require larger AD sizes to run. This will permit binary users to run all inputs packaged with the applications.

## Design

Modify moose/conda/template/build.sh to support the submission of a supplied AD size that is passed via the CIVET recipes that will create the conda package for release. 

## Impact

New feature. Enables running of all inputs packaged in some downstream MOOSE-based apps that rely on larger AD sizes.

### PR DESCRIPTION
**Pull request by milljm**
Add MOOSE_OPTIONS to configure

Allow for MOOSE_OPTIONS to be passed as arguments which will be
used during ./configure of MOOSE.

Closes #26300

### CODE REVIEW

**Comment by GiudGiud** [comment]
MOOSE_CONFIG_OPTIONS? MOOSE_OPTIONS might be too general?

File: scripts/tests/versioner_hashes.yaml [hunk 1]
```diff
@@ -171,3 +171,9 @@ a2e8814e28b97171ff4a06e5343063fed6763dd0: #26102
   libmesh: bad5b94
   moose-dev: 9831cd1
   wasp: 91987a3
+5327cf7e7946b97249ea8f1d65efa86ae24a8e33: #26300
+  mpich: 702900f
+  petsc: 684d2af
+  libmesh: bad5b94
+  moose-dev: 5b104ea
```
Comment (loganharbour) [inline scripts/tests/versioner_hashes.yaml:178]: why did moose-dev change?

Comment (milljm) [reply scripts/tests/versioner_hashes.yaml:178]: Because I modified `moose/conda/moose/*`

**Comment by milljm** [comment]
> MOOSE_CONFIG_OPTIONS? MOOSE_OPTIONS might be too general?

Because that is the option currently being used elsewhere in other Civet recipes, and this _is_ for Packages/Container-like recipes (Conda MOOSE package creation).
```pre
recipes/moosebuild/moose/branch_next/Apptainer_controlled_apps.cfg:MOOSE_OPTIONS = --with-derivative-size=300
```

File: scripts/tests/versioner_hashes.yaml (hunk 1 above)
Comment (milljm) [reply scripts/tests/versioner_hashes.yaml:178]: Hmm, does it change every time with a new commit? I just automatically assumed to run the versioner tool. But I see now thats not necessary... I'll change it back.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**